File: veltahleon/src/main/java/com/testament/veltahleon/model/entities/history/Book.java
package com.testament.veltahleon.model.entities.history;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Entity
@Table(name = "books")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "book_title")
    private String bookName;

    @ManyToOne
    @JoinColumn(name = "author_id")
    private Author authorName;

    @OneToMany(mappedBy = "book")
    private Set<Chapter> chapters;
}

File: frontend/ui/src/app/interfaces/models/history/library/book.ts
// import {Author} from "./author";
// import {Chapter} from "./chapter";

export interface Book {
  id: number;
  name: string;
  authorID: number;
  //author: Author;
  authorFirstName: string;
  authorLastName: string;
  releaseDate: number;
  releaseYearAbbreviation: string;
  //chapters: Chapter[];
  description: string;
  coverURL: string;

  isSelected: boolean;
}
